{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a8d11762",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: mysql-connector-python in c:\\users\\pathu\\anaconda3\\lib\\site-packages (9.1.0)\n",
      "Requirement already satisfied: SQLAlchemy in c:\\users\\pathu\\anaconda3\\lib\\site-packages (1.4.39)\n",
      "Requirement already satisfied: pandas in c:\\users\\pathu\\anaconda3\\lib\\site-packages (1.4.2)\n",
      "Requirement already satisfied: greenlet!=0.4.17 in c:\\users\\pathu\\anaconda3\\lib\\site-packages (from SQLAlchemy) (1.1.1)\n",
      "Requirement already satisfied: pytz>=2020.1 in c:\\users\\pathu\\anaconda3\\lib\\site-packages (from pandas) (2022.1)\n",
      "Requirement already satisfied: numpy>=1.18.5 in c:\\users\\pathu\\anaconda3\\lib\\site-packages (from pandas) (1.21.5)\n",
      "Requirement already satisfied: python-dateutil>=2.8.1 in c:\\users\\pathu\\anaconda3\\lib\\site-packages (from pandas) (2.8.2)\n",
      "Requirement already satisfied: six>=1.5 in c:\\users\\pathu\\anaconda3\\lib\\site-packages (from python-dateutil>=2.8.1->pandas) (1.16.0)\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install mysql-connector-python SQLAlchemy pandas\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "cad3a964",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Connection successful\n"
     ]
    }
   ],
   "source": [
    "from sqlalchemy import create_engine\n",
    "\n",
    "engine = create_engine('mysql+mysqlconnector://root:Pathum5122311@localhost/pathumtest')\n",
    "\n",
    "\n",
    "try:\n",
    "    with engine.connect() as connection:\n",
    "        print(\"Connection successful\")\n",
    "except Exception as e:\n",
    "    print(f\"Error connecting to MySQL: {e}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "95f6e352",
   "metadata": {},
   "outputs": [],
   "source": [
    "create_customers_table = \"\"\"\n",
    "CREATE TABLE IF NOT EXISTS customers (\n",
    "    customer_id VARCHAR(100) PRIMARY KEY,\n",
    "    name VARCHAR(100),\n",
    "    email VARCHAR(100)\n",
    ");\n",
    "\"\"\"\n",
    "\n",
    "create_orders_table = \"\"\"\n",
    "CREATE TABLE IF NOT EXISTS orders (\n",
    "    id VARCHAR(100) PRIMARY KEY,\n",
    "    display_order_id VARCHAR(10),\n",
    "    total_amount DECIMAL(10, 2),\n",
    "    created_at DATETIME,\n",
    "    customer_id VARCHAR(100)\n",
    "    \n",
    ");\n",
    "\"\"\"\n",
    "\n",
    "with engine.connect() as connection:\n",
    "    connection.execute(create_customers_table)\n",
    "    connection.execute(create_orders_table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "40ba73c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "customers_df = pd.read_csv('C:/Users/pathu/Downloads/customers.csv')\n",
    "orders_df = pd.read_csv('C:/Users/pathu/Downloads/order.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "19e1edae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8117"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load customers\n",
    "customers_df.to_sql('customers', con=engine, if_exists='append', index=False)\n",
    "\n",
    "# Load orders\n",
    "orders_df.to_sql('orders', con=engine, if_exists='append', index=False)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
